libpius/path.py

```diff
@@ -0,0 +1,150 @@
+
+import os
+import sys
+import stat
+from os.path import abspath
+import fnmatch
+
+LINUX_TEMPDIR = '/tmp/'
+LINUX_BIN_PATHS = '/usr/bin;/usr/sbin/;/bin;/sbin'
+#binaries is located right in the /usr/local/(program)/ folders
+MAC_BIN_PATHS = '/usr/local'
+
+'''This is grabbed right from the python repo under lib/shutil.py'''
+
+def which(cmd, mode=os.F_OK | os.X_OK, path=None):
+    """Given a command, mode, and a PATH string, return the path which
+    conforms to the given mode on the PATH, or None if there is no such
+    file.
+    `mode` defaults to os.F_OK | os.X_OK. `path` defaults to the result
+    of os.environ.get("PATH"), or can be overridden with a custom search
+    path.
+    """
+    # Check that a given file can be accessed with the correct mode.
+    # Additionally check that `file` is not a directory, as on Windows
+    # directories pass the os.access check.
+    def _access_check(fn, mode):
+        return (os.path.exists(fn) and os.access(fn, mode)
+                and not os.path.isdir(fn))
+
+    # If we're given a path with a directory part, look it up directly rather
+    # than referring to PATH directories. This includes checking relative to the
+    # current directory, e.g. ./script
+    if os.path.dirname(cmd):
+        if _access_check(cmd, mode):
+            return cmd
+        return None
+
+    if path is None:
+        path = os.environ.get("PATH", os.defpath)
+    if not path:
+        return None
+    path = path.split(os.pathsep)
+
+    if sys.platform == "win32":
+        # The current directory takes precedence on Windows.
+        if not os.curdir in path:
+            path.insert(0, os.curdir)
+
+        # PATHEXT is necessary to check on Windows.
+        pathext = os.environ.get("PATHEXT", "").split(os.pathsep)
+        # See if the given file matches any of the expected path extensions.
+        # This will allow us to short circuit when given "python.exe".
+        # If it does match, only test that one, otherwise we have to try
+        # others.
+        if any(cmd.lower().endswith(ext.lower()) for ext in pathext):
+            files = [cmd]
+        else:
+            files = [cmd + ext for ext in pathext]
+    else:
+        # On other platforms you don't have things like PATHEXT to tell you
+        # what file suffixes are executable, so just pass on cmd as-is.
+        files = [cmd]
+
+    seen = set()
+    for dir in path:
+        normdir = os.path.normcase(dir)
+        if not normdir in seen:
+            seen.add(normdir)
+            for thefile in files:
+                name = os.path.join(dir, thefile)
+                if _access_check(name, mode):
+                    return name
+    return None
+#end of "Which" function
+
+def gpg_path():
+    if sys.platform = "win32":
+        gpg = which('gpg2')
+    elif sys.platform = "darwin":
+        #joining all possible paths for location of mac binaries
+        BIN_PATHS = os.path.join(MAC_BIN_PATHS, LINUX_BIN_PATHS)
```

Comment:
os.path.join is going to join with `/` or `\` depending on platform which isn't how you want to join these two things.

Even if it was picking PATHSEP, you'd get `;` on Windows and `:` everywhere else which means on Mac you'd have a non-sensical path of "/usr/local/bin:/usr/bin;/usr/sbin...` of which that only has two elements "/usr/local/bin" and a huge string with semi-colons on it, and on Linux you'd have no working path at all.

Reply:
whoops meant to add them with a semicolon inbetween